### PR TITLE
feat(auth-and- unauth-guard)

### PR DESCRIPTION
**Pull request by YgRamos9**
## Summary
- add guard auth and unauth

## Screenshot
https://user-images.githubusercontent.com/107204529/183765957-1fafafc5-ef69-4f67-a82d-00c6f2a10f03.mov

Captura de Pantalla 2022-08-10 a la(s) 9 36 35 a m

## Trello Card
https://trello.com/c/gwvNsvEt/8-auth-guard-unauth-guard

### CODE REVIEW

**Comment by miguelhenaoa** [comment]
@YgRamos9, before reviewing the PR, add the coverage folder to the .gitignore file, for a more efficient review.

File: angular/src/app/components/form-container/form-container.component.ts [hunk 1]
```diff
@@ -8,9 +8,9 @@ import { IFields, IUser } from 'src/app/interfaces/global.interface';
   styleUrls: ['./form-container.component.scss'],
 })
 export class FormContainerComponent {
+  @Output() handleOnSubmit = new EventEmitter<IUser>();
   @Input() formFields!: IFields[];
   @Input() buttonSend!: string;
-  @Output() handleOnSubmit = new EventEmitter<IUser>();
   @Input() errorMessage!: string;
   @Input() formGroup!: FormGroup;
```
Comment (miguelhenaoa) [inline angular/src/app/components/form-container/form-container.component.ts]: The order of the decorators is Input first, then Output.

File: angular/src/app/guards/auth.guard.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { Injectable } from '@angular/core';
+import {
+  ActivatedRouteSnapshot,
```
Comment (miguelhenaoa) [inline angular/src/app/guards/auth.guard.ts]: Delete unused imports

File: angular/src/app/guards/unauth.guard.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { Injectable } from '@angular/core';
+import {
+  ActivatedRouteSnapshot,
```
Comment (miguelhenaoa) [inline angular/src/app/guards/unauth.guard.ts]: Same

File: angular/src/app/guards/unauth.guard.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  RouterStateSnapshot,
+  UrlTree,
+} from '@angular/router';
+import { Observable } from 'rxjs';
+import { UserService } from '../services/user.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class UnauthGuard implements CanActivate {
```
Comment (miguelhenaoa) [inline angular/src/app/guards/unauth.guard.ts]: Rename UnAuthGuard

**Review by miguelhenaoa — CHANGES_REQUESTED** [review]

File: angular/src/app/services/user.service.ts [hunk 5]
```diff
@@ -41,4 +41,8 @@ export class UserService {
   logOut() {
     localStorage.removeItem('userToken');
   }
+
+  isLogin() {
```
Comment (joacomiguens) [inline angular/src/app/services/user.service.ts]:
```suggestion
  isLogged() {
```

File: angular/src/app/screens/auth/screens/book-list/book-list.component.ts [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+
+@Component({
+  selector: 'app-book-list',
+  templateUrl: './book-list.component.html',
+  styleUrls: ['./book-list.component.scss'],
+})
+export class BookListComponent implements OnInit {
+  constructor() {}
+
+  ngOnInit(): void {}
```
Comment (joacomiguens) [inline angular/src/app/screens/auth/screens/book-list/book-list.component.ts]: Delete unused methods please

File: angular/src/app/screens/auth/screens/book-list/book-list.component.scss [hunk 7]
```diff
@@ -0,0 +1 @@
+
```
Comment (joacomiguens) [inline angular/src/app/screens/auth/screens/book-list/book-list.component.scss]: delete file

**Review by joacomiguens — APPROVED** [review]

**Review by miguelhenaoa — APPROVED** [review]
LGTM!! 🚀

**Pull request merged**